File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] answer = new int[nums.length];
        Arrays.fill(answer,1);
        
        int prev = 1;
        for(int i=1;i<nums.length;i++)
        {
            prev = prev*nums[i-1];
            answer[i] = prev;
        }

        prev = nums[nums.length-1];

        for(int i= nums.length-2;i>=0;i--)
        {
            answer[i] = answer[i]*prev;
            prev = prev*nums[i];
        }

        return answer;
    }
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        output = []
        prev = 1
        for i in range(0,len(nums)):
            output.append(prev)
            prev = prev*nums[i]
        
        prev = nums[len(nums)-1]
        for i in range(len(nums)-2,-1,-1):
            output[i] *=prev
            prev = prev*nums[i]
        

        return output
        